servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -277,18 +277,17 @@ public void onComplete() {
                     // Invoke the selector before adding the connection to the pool, otherwise, connection can be used
                     // concurrently and hence a new connection can be rejected by the selector.
                     if (!selector.test(newCnx)) {
-                        newCnx.closeAsync().subscribe();
                         // Failure in selection could be temporary, hence add it to the queue and be consistent with the
                         // fact that select failure does not close a connection.
-                        return failed(new ConnectionRejectedException("Newly created connection " + newCnx +
-                                " rejected by the selection filter."));
+                        return newCnx.closeAsync().concat(failed(new ConnectionRejectedException(
```

Comment:
Can you verify if we have a test that checks the connection is closed when we get the error?